File: fiber_bundles/bindings/java/examples/read_t.java

//
//
//
// Copyright (c) 2013 Limit Point Systems, Inc.
//
//
// Class read_t
//

import bindings.java.*;

public class read_t extends base_example
{

  static final String moduleName = "fiber_bundles_java_binding";
  static final String className  = "read_t";

  public read_t()
  {
    super(moduleName);
  }

  public read_t(String module_name)
  {
    super(module_name);
  }

  public static void usage()
  {
    String nl = System.getProperty("line.separator");

    StringBuffer sb = new StringBuffer();
    sb.append("Usage: java [-cp CLASSPATH] " + className
              + " hdf_file_name").append(nl);

    System.out.println(sb);
    System.exit(1);
  }

  public static void main(String[] args)
  {

    // Handle command line arguments.

    String fileName = null;

    if(args.length < 1)
    {
      usage();
    }
    else
    {
      fileName = args[0];
    }

    // Instantiate "this" class (loads the libraries, etc.).

    read_t test = new read_t();

    // Make the namespace poset.
    // (Note: We have to supply the default args in Java).

    namespace_poset the_namespace =
      new namespace_poset("test", 8, 16, 4);

    // Read the namespace poset from the file.

    test.readHdfFile(the_namespace, fileName);

    // Output a text version to stdout.

    System.out.println(the_namespace.to_string());

  }
}

File: sheaves/bindings/python/examples/read_t.py
#! /usr/bin/env python
##
## Copyright (c) 2013 Limit Point Systems, Inc.
##


import sys ## for command line arguments
from sheaves_python_binding import *

TRUE = 1
FALSE = 0

###############################################################################
# read_namespace(file_name):
###############################################################################

def read_namespace(file_name):

    ## Make the default namespace.

    ns = namespace_poset(file_name[0:-4]) ## Splice off ".hdf" from file_name

    ## Read the namespace from the file;
    ## need write access in order to read it.

    ns.get_read_write_access()
    
    sa = storage_agent(file_name, sheaf_file.READ_ONLY, FALSE)
    sa.read_entire(ns)
    
    return ns

###############################################################################
# main(file_name):
###############################################################################

def main(file_name):

    ## Read the namespace from the file.
    
    ns = read_namespace(file_name)
    
    ## Output a text version to stdout.
    
    ns.to_stream()

###############################################################################

if __name__ == '__main__':  
    import sys   
    if len(sys.argv) < 2:
        print 'Usage: ' + sys.argv[0] + ' file_name' 
    else:
        main(sys.argv[1])
